packages/fe-fpm-writer/src/filter/index.ts

```diff
@@ -0,0 +1,107 @@
+import { create as createStorage } from 'mem-fs';
+import type { Editor } from 'mem-fs-editor';
+import { create } from 'mem-fs-editor';
+import type { CustomFilter, InternalCustomFilter } from './types';
+import { join } from 'path';
+import { render } from 'ejs';
+import { validateBasePath } from '../common/validate';
+import type { Manifest } from '../common/types';
+import { setCommonDefaults } from '../common/defaults';
+import { getTemplatePath } from '../templates';
+import { getJsonSpace } from '../common/file';
+import { applyEventHandlerConfiguration, contextParameter } from '../common/event-handler';
+import type { FilterField } from '../building-block/types';
+
+/**
+ * Enhances the provided custom filter configuration with default data.
+ *
+ * @param {CustomFilter} data - a custom filter configuration object
+ * @param {string} manifestPath - path to the project's manifest.json
+ * @param {Manifest} manifest - the application manifest
+ * @returns enhanced configuration
+ */
+function enhanceConfig(data: CustomFilter, manifestPath: string, manifest: Manifest): InternalCustomFilter {
+    // clone input
+    const config: CustomFilter & Partial<InternalCustomFilter> = data;
+
+    setCommonDefaults(config, manifestPath, manifest);
+
+    // set default values for requirement, language, the fragment file name
+    config.required = config.required || false;
+    config.typescript = !!config.typescript;
+    config.fragmentFile = config.fragmentFile || config.name;
+    config.eventHandlerFnName =
```

Comment:
I am wondering if we really need this property, couldn't you do this with the existing `config.eventHandler.fnName`? E.g.
```
if (config.eventHandler === true) {
   config.eventHandler = {};
}

if (typeof config.eventHandler === 'object' && !config.eventHandler.fnName) {
   config.eventHandler.fnName = 'itemsFilter;
}
```
As result `applyEventHandlerConfiguration` should later behave exactly is it does now.

Reply:
Fixed with #1027